lib/rules/sort-comp.js

```diff
@@ -58,7 +57,7 @@ const defaultConfig = {
 function getMethodsOrder(userConfig) {
   userConfig = userConfig || {};
 
-  const groups = util._extend(defaultConfig.groups, userConfig.groups);
+  const groups = Object.assign(defaultConfig.groups, userConfig.groups);
```

Comment:
this should probably not mutate the default groups - this might be a bug in the rule itself.
```suggestion
  const groups = Object.assign({}, defaultConfig.groups, userConfig.groups);
```